### ISSUE TITLE
Can we retrieve the raw XML from a request rather than an object.

### ISSUE THREAD

**Issue by James-Parsons** [AuthorizeNet/sdk-php#259]
I have several functions that look like this:

public function getCustomerProfiles() {
	$customerProfiles = array();

	// Before we can get customer profiles, we need to get a list of all customer id's.
	$customerIdListRequest = new AnetAPI\GetCustomerProfileIdsRequest();
	$customerIdListRequest->setMerchantAuthentication(self::getMerchantAuth(Config::LOGIN_ID, Config::TRANSACTION_KEY));

	$customerIdListController = new AnetController\GetCustomerProfileIdsController($customerIdListRequest);
	$customerIdListResponse = $customerIdListController->executeWithApiResponse(\net\authorize\api\constants\ANetEnvironment::SANDBOX);

	if(($customerIdListResponse != null) && ($customerIdListResponse->getMessages()->getResultCode() == "Ok")) {
		foreach( $customerIdListResponse->getIds() as $id ) {
			// Now we can get each customer profile.
			$request = new AnetAPI\GetCustomerProfileRequest();
			$request->setMerchantAuthentication(self::getMerchantAuth(Config::LOGIN_ID, Config::TRANSACTION_KEY));
			$request->setCustomerProfileId($id);

			$controller = new AnetController\GetCustomerProfileController($request);
			$response = $controller->executeWithApiResponse(\net\authorize\api\constants\ANetEnvironment::SANDBOX);

			if(($response != null) && ($response->getMessages()->getResultCode() == "Ok")) {
				// Add it to the array.
				array_push($customerProfiles, $response->getProfile());
			} else {
				throw new \Exception($response->getMessages()->getMessage());
			}

		}
	} else {
		throw new \Exception($customerIdListResponse->getMessages()->getMessage());
	}

	return $customerProfiles;
    }

Right now, I'm just retrieving objects to an array. Can I use the SDK to retrieve a raw response?

**Comment by kikmak42** [comment]
Hi @James-Parsons 

This SDK is designed to ease developers in using the APIs without requiring to handle the request and response manually. But as you require the XML response, the best way would be to make direct calls to API using curl.

Or you can skip de-serialization of the xml to objects in [ApiOperationbase.php](https://github.com/AuthorizeNet/sdk-php/blob/master/lib/net/authorize/api/controller/base/ApiOperationBase.php#L129) and pass the xml response object directly to the caller.

**Comment by ashtru** [comment]
Thank you for raising this issue. Hope this answers your specific requirement. Please raise a new issue, if you’re still facing problems, and we’ll look into it quickly.
